### DISCUSSION TITLE
Bearer Authentication?

### DISCUSSION THREAD

**Discussion by stevenroman760** [unoplatform/uno#9267] [General]
Hello,

I inherited some code, this code works on a standard windows app for obtaining a token but when I run thru the emulator for Android the code below just hangs. Are we not allowed to request a token from a mobile app?

                IConfidentialClientApplication app;

                app = ConfidentialClientApplicationBuilder.Create(config.ClientId)
                    .WithClientSecret(config.ClientSecret)
                    .WithAuthority(new Uri(config.Authority))
                    .Build();

Thanks,
Steven

**Comment by nickrandolph** [comment]
I think that @carldebilly has answered this question here: https://github.com/unoplatform/uno/discussions/8555
In summary ConfidentialClientApplicationBuilder isn't supported. You need to use PublicClientApplicationBuilder
